s3/compress.go

```diff
@@ -233,13 +234,21 @@ func (c *Compressor) Upload(ctx context.Context, body io.Reader, bucket, key *st
 func (c *Compressor) CreateObject(ctx context.Context, db models.DBInterface, body io.Reader, urlparams *models.URLParams, payload *models.ExportPayload) error {
 	filename := fmt.Sprintf("%s/%s/%s.%s", payload.OrganizationID, payload.ID, urlparams.ResourceUUID, payload.Format)
 
+	var failedUploads = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "failed_uploads_total",
+		Help: "The total number of failed S3 uploads.",
+	})
+
+	prometheus.MustRegister(failedUploads)
+
 	if err := payload.SetStatusRunning(db); err != nil {
 		c.Log.Errorw("failed to set running status", "error", err)
 		return err
 	}
 
 	_, uploadErr := c.Upload(ctx, body, &c.Bucket, &filename)
 	if uploadErr != nil {
+		defer failedUploads.Inc()
```

Comment:
No need for a `defer` here.  The `defer` keyword makes it so that the function is not called immediately.  The compiler/runtime schedules the deferred function to be called once we leave the scope.  In this case, we want `failedUploads.Inc()` to be called immediately.